### ISSUE TITLE
intro-to-pytorch/Part 8 - Transfer Learning (Solution) -> Entire validation loop is run after every 5 training batches. This wastes compute & is an inefficient way to calculate accuracy

### ISSUE THREAD

**Issue by jaintj95** [udacity/deep-learning-v2-pytorch#148]
In the last cell (solution to the exercise)
print_every is set to 5
After passing just 5 batches of training data through the network, the entire validation (test) loop is run (40 batches of validation data). This happens after every 5 training batches.

5 batches wouldn't be enough for the network to generalise and perform well on validation data. The standard way of testing on validation dataset after all training batches and increasing the number of epochs would be a better idea.

Current implementation wastes time and compute. Why would we calculate accuracy like that? Also, this teaches wrong validation technique to the students. I remember a lot of students were confused and stuck using this technique to calculate accuracy in lab project of PyTorch Challenge.

**Comment by jaintj95** [comment]
![transfer_learning](https://user-images.githubusercontent.com/36858630/53390464-3d330e80-39b9-11e9-85ce-a10512b1cdc7.jpg)
I just added print statements for no. of training and validation batches. I hope the issue is clearer now.

**Comment by iqDF** [comment]
Well certainly I think the blocks for `# Calculate accuracy` is a bit confusing for readers. I would rather use one line `np.mean(top_class == labels)` to calculate acc directly as it is more readble. Also can @cezannec give us the idea of using `batch_size=5`?

**Comment by abhiojha8** [comment]
Thanks for the feedback. We will address this in future refreshes.
